models/timm.py

```diff
@@ -15,26 +15,20 @@
 import torch
 import torch.nn as nn
 
-
 PYTORCH_IMAGE_MODELS_REPO = 'rwightman/pytorch-image-models'
 
-class EfficientNet(nn.Module):
-    def __init__(self, backbone='tf_efficientnet_b0_ns', pretrained=True):
-        super(EfficientNet, self).__init__()
+class PyTorchImageModels(nn.Module):
+    def __init__(self, backbone='resnet50d', pretrained=True):
+        super(PyTorchImageModels, self).__init__()
         self.encoder = torch.hub.load(
             PYTORCH_IMAGE_MODELS_REPO,
             backbone, 
-            pretrained=pretrained,
+            pretrained,
         )
-        self.in_features = self.encoder.classifier.in_features
+        for m in self.encoder.modules():
+            if isinstance(m, (nn.Linear)):
+                self.in_features = m.in_features
```

Comment:
I believe that all timm models support the following syntax to get the number of features produced by the models:

```
self.in_features = self.encoder.num_features
```